src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs

```diff
@@ -295,7 +295,11 @@ public void Paint(Rect rect)
             CompositionTarget.ImmediateUIThreadRender();
     }
 
-    public void Start() => CompositionTarget.IsEnabled = true;
+    public void Start()
+    {
+        CompositionTarget.IsEnabled = true;
+        CompositionTarget.ImmediateUIThreadRender();
```

Comment:
This is not guaranteed to be supported on all platforms.

Reply:
@kekekeks any suggestions what to do then?

Reply:
@kekekeks I believe I fixed it, moved the check to every place before the `Commit().Wait()` by checking the `IsEnabled` property of the `ServerCompositionTarget`.
This should make sure there are no deadlocks, but I am still not familiar with other platforms, so I will need your feedback on that.